### DISCUSSION TITLE
What is the right way to extend shield session?

### DISCUSSION THREAD

**Discussion by mel9pr1** [codeigniter4/shield#1208] [Q&A]
I want to make some changes in a method in Shield **Session.php** so I put a Session.php file at the Codeigniter's **App\Libraries\Shield** folder with a session file with this;

```php
namespace App\Libraries\Shield;

use CodeIgniter\Shield\Authentication\Authenticators\Session as ShieldSession;

class Session extends ShieldSession
{
     public function attempt(array $credentials): Result
    {
		// modification here....
     }
}
```` 

Do I missing something to make it work?

Thanks!

**Comment by Ty120301** [comment]
I guess it should if you use it and update the App/Config/Auth.php too?
`public array $authenticators = [];`

**Comment by warcooft** [comment]
at the moment, this does not seem possible, see https://github.com/codeigniter4/shield/discussions/1197#discussion-7153498

**Comment by grimpirate** [comment]
Definitely possible as I've done it in for my own extension [halberd](https://github.com/grimpirate/halberd/blob/develop/src/Authentication/Authenticators/TOTP.php). It doesn't replace the default Session Authenticator if that is what you're attempting to do. In that case you would have to replace Auth.authenticators.session with your custom Session authenticator.
